docs/introduction/entities.md

```diff
@@ -0,0 +1,285 @@
+
+    /**
+     * @param string|null $seoTitle
+     */
+    public function setSeoTitle($seoTitle)
+    {
+        $this->seoTitle = $seoTitle;
+    }
+
+    // ...
+
+}
+```
+
+...and its main entity `Brand` working as a proxy:
+
+```php
+// FrameworkBundle/Model/Product/Brand/Brand.php
+
+namespace Shopsys\FrameworkBundle\Model\Product\Brand;
+
+/**
+ * @ORM\Table(name="brands")
+ * @ORM\Entity
+ */
+class Brand extends AbstractTranslatableEntity
+{
+
+    // ...
+
+    /**
+     * @param int $domainId
+     * @return string
+     */
+    public function getSeoTitle(int $domainId)
+    {
+        return $this->getBrandDomain($domainId)->getSeoTitle();
+    }
+    
+    /**
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Model\Product\Brand\BrandDomain
+     */
+    protected function getBrandDomain(int $domainId)
+    {
+        foreach ($this->domains as $domain) {
+            if ($domain->getDomainId() === $domainId) {
+                return $domain;
+            }
+        }
+
+        throw new BrandDomainNotFoundException($this->id, $domainId);
+    }
+
+    // ...
+
+}
+```
+
+## Entity data
+
+Is a data object that is used to transfer data through application and also to create an entity.
+The entity data can be created in the UI (or other data source), then propagated via facade and finally used to create the entity.
+The entity data can be also created from an entity, and propagated to the UI or other parts of application.
+
+Entity data have all attributes public and is mutable.
+Entity data have a constructor without parameters and all parameters are initialized in the constructor.
+
+### Example
+
+```php
+// FrameworkBundle\Model\Product\Brand\BrandData.php
+
+namespace Shopsys\FrameworkBundle\Model\Product\Brand;
+
+use Shopsys\FrameworkBundle\Component\FileUpload\ImageUploadData;
+use Shopsys\FrameworkBundle\Form\UrlListData;
+
+class BrandData
+{
+    /**
+     * @var string
+     */
+    public $name;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\FileUpload\ImageUploadData
+     */
+    public $image;
+
+    /**
+     * @var string[]
+     */
+    public $descriptions;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Form\UrlListData
+     */
+    public $urls;
+
+    /**
+     * @var string[]|null[]
+     */
+    public $seoTitles;
+
+    /**
+     * @var string[]|null[]
+     */
+    public $seoMetaDescriptions;
+
+    /**
+     * @var string[]|null[]
+     */
+    public $seoH1s;
+
+    public function __construct()
+    {
+        $this->name = '';
+        $this->image = new ImageUploadData();
+        $this->descriptions = [];
+        $this->urls = new UrlListData();
+        $this->seoTitles = [];
+        $this->seoMetaDescriptions = [];
+        $this->seoH1s = [];
+    }
+}
+```
+
+... and example of entity construction
+
+```php
+$brandData = new BrandData();
+// $brandData->name = ...
+// ...
+$brand = $this->brandFactory->create($brandData);
+```
+
+## Entity data factory
```

Comment:
It would be nice to mention `ObjectIsCreatedByFactorySniff` somewhere

Reply:
Hm... the same is valid also for entity factory. Shouldn't I mention it in both? But it can be strange to have the same sentence twice in text... what do You think?